### PR TITLE
[feat] add ledger compiler support

### PR DESCRIPTION
**Pull request by ramsayleung**
## Description

目前 `double-entry-generator` 只支持 `beancount` 后端，增加对 `ledger` 后端的支持.  针对不同的 `provider`, 后端需要有对应的输出模板，已有的输出模板包括:

- [x] 普通消费账单的模板
- [ ] 火币模板1
- [ ] 火币模板2
- [ ] 火币卖出模板
- [ ] 海通买入模板

本次PR 先增加 `普通消费账单的模板` 的支持

## Modification
概括:

1. 增加 `compiler/ledger` package
2. `test` 目录增加 `ledger` 相关test script
3.  `test` 目录，重命名已有的 script, 增加 `beancount`  后缀
4.  `example` 目录增加 `ledger` 相关的生成文件.

## Motivation and Context

#92 

## Dependencies 

None

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How has this been tested?

新增三个 end-to-end 测试脚本, 测试 `ledger` 后端生成功能:

- [x] alipay-test-ledger.sh: pass
- [x] wechat-test-ledger.sh: pass
- [x] icbc-test-ledger.sh: pass

对生成的 ledger 文件进行语法与格式校验:

- [x] `ledger -f test/output/test-icbc-credit-output.ledger bal`: pass
- [x] `ledger -f test/output/test-icbc-debit-output.ledger bal`: pass
- [x] `ledger -f test/output/test-alipay-output.ledger bal`: pass
- [x] `ledger -f test/output/test-wechat-output.ledger bal`: pass   

`make test`: all tests pass

### CODE REVIEW

**Review by Triple-Z — CHANGES_REQUESTED** [review]
It's a pretty good PR, please make a few minor changes.

**Comment by Triple-Z** [comment]
PTAL @gaocegege

File: test/alipay-test-ledger.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (ramsayleung) [reply test/alipay-test-ledger.sh:1]: fixed :)

File: Makefile [hunk 2]
```diff
@@ -96,19 +96,22 @@ test-go: ## Run Golang tests
 	@go test ./...
 
 test-alipay: ## Run tests for Alipay provider
```
Comment (Triple-Z) [inline Makefile:98]: `test-alipay-beancount` and `test-alipay-ledger` or just `test-alipay`?

I prefer the first ones, what do u guys think? /cc @gaocegege @ramsayleung

Comment (ramsayleung) [reply Makefile:98]: Sure, it makes sense to me either.

File: Makefile [hunk 3]
```diff
@@ -90,25 +90,32 @@ clean: ## Clean all the temporary files
 	@rm -rf ./double-entry-generator
 	@rm -rf ./wasm-dist
 
-test: test-go test-alipay test-wechat test-huobi test-htsec test-icbc ## Run all tests
+test: test-go test-alipay-beancount test-alipay-ledger test-wechat-beancount test-wechat-ledger \
+	  test-huobi-beancount test-htsec-beancount test-icbc-beancount test-icbc-ledger ## Run all tests
```
Comment (Triple-Z) [inline Makefile:94]: The comment should at the target line (I guess), or it won't print help for `test` target when using `make help`.

**Review by Triple-Z — APPROVED** [review]
LGTM.

File: Makefile (hunk 3 above)
Comment (ramsayleung) [reply Makefile]: Thanks for your heads-up, I wasn't aware of that before.

**Comment by Triple-Z** [comment]
Thanks for your contribution 🎉

**Pull request merged**